### PR TITLE
Reorder parameters in apache rewrite rule and add + sign in token regex

### PR DESCRIPTION
**Pull request by arnaud-morvan**

### CODE REVIEW

File: tilecloud_chain/controller.py [hunk 1]
```diff
@@ -387,15 +388,24 @@ def _generate_apache_config(gene):
...
-                        '%(dimensions_rep)s%(zoom)s/%(final)s [PT]\n' % {
+                        'RewriteRule'
+                        ' '
+                        '^%(tiles_location)s/1.0.0/%(layer)s/%(token_regex)s/' # Baseurl/layer/Style
+                        '%(dimensions_re)s/'  # Dimensions : variable number of values
+                        '/%(token_regex)s/%(zoom)s/(.*)$' # TileMatrixSet/TileMatrix/TileRow/TileCol.extension
+                        ' '
+                        '%(mapcache_location)s/wmts/1.0.0/%(layer)s/$1'
+                        '/%(dimensions_rep)s'
+                        '/$%(dim+2)s/%(zoom)s/$%(dim+3)s'
```
Comment (sbrunner) [inline tilecloud_chain/controller.py]: As I tested dim+2 and dim+3 will not works ...

File: tilecloud_chain/controller.py [hunk 2]
```diff
@@ -379,6 +379,7 @@ def _generate_apache_config(gene):
         if not gene.validate_mapcache_config():
             exit(1)  # pragma: no cover
     if use_mapcache and not use_server:
+        token_regex = '([a-zA-Z0-9_\-\+~\.]+)/'
```
Comment (sbrunner) [inline tilecloud_chain/controller.py]: Shoud not include the '/', or should be removed after ...

File: tilecloud_chain/controller.py [hunk 3]
```diff
@@ -387,15 +388,24 @@ def _generate_apache_config(gene):
...
+                        ' '
+                        '[PT]\n' % {
                             'tiles_location': gene.config['apache']['location'],
                             'mapcache_location': gene.config['mapcache']['location'],
                             'layer': layer['name'],
-                            'dimensions_re': ''.join(['([a-zA-Z0-9_\-~\.]+)/' for e in range(dim)]),
-                            'dimensions_rep': ''.join(['$%i/' % (e + 3) for e in range(dim)]),
-                            'final': '$%i' % (3 + dim),
+                            'param_regex': token_regex,
+                            'dimensions_re': '/'.join([token_regex for e in range(dim)]),
```
Comment (sbrunner) [inline tilecloud_chain/controller.py]: You moved the '/' position than it will no more works without any dimensions ...

**Comment by arnaud-morvan** [comment]
Thanks for review, I was testing with another version in my GMF virtual env.
Now it should works.

**Comment by sbrunner** [comment]
Nice now it should works except when we don't have any dimensions, because we will have two '/' where we place the dimensions...

**Comment by arnaud-morvan** [comment]
Good catch

**Comment by sbrunner** [comment]
Should be rebase to make the tests pass :-)

**Comment by arnaud-morvan** [comment]
Done

**Comment by sbrunner** [comment]
In the Travis test we can see two code style error and 3 test failed (open the line 1226) than I hope that's done :-)

**Comment by arnaud-morvan** [comment]
Travis is happy now. I think you could merge.

File: tilecloud_chain/controller.py [hunk 4]
```diff
@@ -387,15 +388,25 @@ def _generate_apache_config(gene):
...
                             'mapcache_location': gene.config['mapcache']['location'],
                             'layer': layer['name'],
-                            'dimensions_re': ''.join(['([a-zA-Z0-9_\-~\.]+)/' for e in range(dim)]),
-                            'dimensions_rep': ''.join(['$%i/' % (e + 3) for e in range(dim)]),
-                            'final': '$%i' % (3 + dim),
+                            'token_regex': token_regex,
+                            'dimensions_re': ''.join(['/'+token_regex for e in range(dim)]),
+                            'dimensions_rep': ''.join(['/$%i' % (e + 2) for e in range(dim)]),
+                            'tilematrixset': dim+2,
+                            'final': dim+3,
```
Comment (sbrunner) [inline tilecloud_chain/controller.py]: Tree times missing spaces around the `+` operator

**Comment by sbrunner** [comment]
Just a small comment, fix it and I will merge it :-)

**Comment by arnaud-morvan** [comment]
Done, missing spaces added.

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**